Lib/unittest/mock.py

```diff
@@ -2690,11 +2695,12 @@ def create_autospec(spec, spec_set=False, instance=False, _parent=None,
         # should only happen at the top level because we don't
         # recurse for functions
         mock = _set_signature(mock, spec)
-        if is_async_func:
-            _setup_async_mock(mock)
     else:
         _check_signature(spec, mock, is_type, instance)
 
+    if is_async:
+        _setup_async_mock(mock)
```

Comment:
The actual await functionality doesn't quite work yet because the existing AsyncMock assumes it's a coroutine where the awaitable class case requires a non-coroutine instead. Still toying around with ways of getting around this issue but, until then, the current implementation won't fully solve this issue.